### PR TITLE
Compatibility change for IBC

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
- Change data type of `key` and `value` for `EventAttribute` and change all references
- Add missing fields to ABCI request structs

## Testing performed to validate your change
establish connection with axelar

### CODE REVIEW

File: rpc/coretypes/responses.go [hunk 1]
```diff
@@ -241,7 +241,9 @@ type ResultConsensusState struct {
 type ResultBroadcastTx struct {
 	Code      uint32         `json:"code"`
 	Data      bytes.HexBytes `json:"data"`
+	Log       string         `json:"log"`
 	Codespace string         `json:"codespace"`
+
```
Comment (philipsu522) [inline rpc/coretypes/responses.go:246]: nit - remove space

File: rpc/coretypes/responses.go [hunk 2]
```diff
@@ -241,7 +241,9 @@ type ResultConsensusState struct {
 type ResultBroadcastTx struct {
 	Code      uint32         `json:"code"`
 	Data      bytes.HexBytes `json:"data"`
+	Log       string         `json:"log"`
```
Comment (philipsu522) [inline rpc/coretypes/responses.go:244]: is this needed?

**Review by philipsu522 — APPROVED** [review]

File: rpc/coretypes/responses.go (hunk 2 above)
Comment (codchen) [reply rpc/coretypes/responses.go:244]: @LCyson do you remember whether this was added for debugging or needed for ibc to work?

Comment (LCyson) [reply rpc/coretypes/responses.go]: Yes this is needed, otherwise hermes will break bc it couldn't find the log

**Review by LCyson — APPROVED** [review]

File: rpc/coretypes/responses.go (hunk 2 above)
Comment (LCyson) [reply rpc/coretypes/responses.go]: hm it seems that we removed this^ we prob want to add it back

**Pull request merged**